src/bitcoind_client.rs

```diff
@@ -47,8 +51,19 @@ impl BitcoindClient {
 
 	}
 
-	pub async fn verifytxoutproof() {
+	pub async fn verifytxoutproof(txid: String, slot: usize, mut inclusion_proof: InclusionProof) -> bool {
```

Comment:
I think this is where we’re out-of-sync:

`verifytxoutproof` is a RPC method in Bitcoin Core: https://github.com/bitcoin/bitcoin/blob/master/src/rpc/txoutproof.cpp#L123

This method takes a `CMerkleBlock` and then return a result if the pointed to `txid` has been included in the chain. `CMerkleBlock` contains a `CPartialMerkleTree` (which internally points to a txid).

Here `verifytxoutproof` should be just a call to `self.rpc_client.call(“verifytxoutproof”, &[merkle_block])` to realize the latest step of mainstay proof verification, namely that the txid has been included in the chain.

Now in term of API, we have two options (at least):
- a) move this verification code (commitment, slot proof, merkle root inclusion) in the client-side (i.e into `civkit-sample`)
- b) have just the client with a method `verifiyinclusionproof` and a given txid

As the ultimate step of the verification flow is dependent on chain access and txidindex (which takes some server ressource), mobile clients will most of the time rely on a server. So I think we can have this code here, just wrapping `verify_commitments` / `verify_slot_proof` and `verify_merkle_root_inclusion` in a dedicated mainstay method and have this in `notaryd` calling back `verifytxoutproof` for the latest step).

Reply:
Want to clarify what we want to do here and how. 

In order to verify the merkle root inclusion in the mainstay transaction we need to get the `scriptPubKey` of the tx and to verify it's confirmation. The current method is that the mainstay proof just contains the merkle root (and path) and the TxID, and the verification process consists of using `getrawtransaction` to confirm the txid is confirmed AND get the `scriptPubKey` from the raw/deserialised tx returned (then verify `scriptPubKey` is tweaked with the merkle root). 

This is the simplest approach as it only means the mainstay 'proof' object needs the TxID and merke root, however it also requires a node with `txindex=1`. 

If we want to enable verification of proofs with a pruned node (with `txindex=0`) then we can use `verifytxoutproof `. But for this to work, we need to know the a) The full raw transaction and b) The proof generated by `gettxoutproof `. These would have to be included in the 'proof' object. 

`gettxoutproof` requires `txindex=1` so a a pruned node would only suffice for verification, you would need a full node for generating proofs. 

Does this make sense - we should go with this?

Reply:
> If we want to enable verification of proofs with a pruned node (with txindex=0) then we can use verifytxoutproof . But for this to work, we need to know the a) The full raw transaction and b) The proof generated by gettxoutproof . These would have to be included in the 'proof' object.

I think we’re in sync here - Note to call `verifytxoutproof` you only need the merkle block generated by `gettxoutproof`, the full raw transaction isn’t a requirement, see the tests in `rpc_txoutproof.py`.

That said, of course better to use `getrawtransaction` if this fits the current usage of verifying the merkle root inclusion in a mainstay transaction. Note you can use `getrawtransaction` with the current civkitd rpc framework with `Client::call()`.

Note my feedback was mostly on the resource trade-offs and verification responsibility between client and server. As both alternatives are requiring `txindex=1`, I don’t think it matters. I can see a new `verifyinclusionproof` with the scriptpubkey, the txid and the merkle root, the commitment and the merkle path. What is unclear to me is how the user learns the scriptpubkey, from our conversation here: https://github.com/civkit/civkit-node/pull/64#discussion_r1340345073

I think we should go with this, even if the high-level API we should introduce for the civkit user at the client-level (i.e currently `civkit-sample` is unclear to me, we can address this in a follow-up PR.

Reply:
The user can only learn the scriptPubKey from the raw (deserialised) tx - that's why it needs to be included in the proof if can't be retrieved with `getrawtransaction `.